src/fondant/pipeline/pipeline.py

```diff
@@ -704,11 +703,16 @@ def apply(
             image = ref.image()
             description = ref.__doc__ or "python component"
 
+            consumes_spec = {k: v.type.to_json() for k, v in self.fields.items()}
+            if consumes:
+                for k, v in consumes.items():
+                    consumes_spec[k] = consumes_spec[v]
+
```

Comment:
This needs more checks and might have to be moved to a better place. Just wanted to get a PoC working.